include/circt/Dialect/RTL/Ops.h

```diff
@@ -12,26 +12,33 @@
 #include "mlir/IR/OpImplementation.h"
 #include "mlir/IR/RegionKindInterface.h"
 #include "mlir/IR/SymbolTable.h"
+#include "mlir/Interfaces/ControlFlowInterfaces.h"
 #include "mlir/Interfaces/SideEffectInterfaces.h"
 
 namespace circt {
 namespace rtl {
 
-// This holds the name, type, direction of a module's ports
+/// A RTL module ports direction.
+enum PortDirection {
+  INPUT = 1,
+  OUTPUT = 2,
+  INOUT = 3,
+};
+
+/// This holds the name, type, direction of a module's ports
 struct RTLModulePortInfo {
   StringAttr name;
+  PortDirection direction;
   Type type;
-  StringAttr direction;
+  size_t argNum; // Either the argument index or the result index depending on
```

Comment:
Does it work to declare this as `size_t argNum = ~0U;` to reduce the chance of an uninitialized variable when working with RTLModulePortInfo?

Reply:
Like a sentinel value which I could check for via asserts?

Now that you mention it, I'm tempted to delete the default constructor and add a constructor for this reason. Too far?

Reply:
I'd keep the default ctor (it is used when filling it in element wise) but just init it with something that is obvious in a debugger if it gets forgotten.